### ISSUE TITLE
3.3.0 breaks latest Chrome

### ISSUE THREAD

**Issue by adrienharnay** [luisrudge/postcss-flexbugs-fixes#45]
Hey,

I understand why #43 has been done, but since upgrading to 3.3.0, all my layouts are broken in latest Chrome. Flex elements overlap over each other in a big mush. Should the specs be followed on this matter?

Thanks!

**Comment by luisrudge** [comment]
can you create a codepen so I can take a look?

**Comment by zslabs** [comment]
☝️ Same. If I write `flex: 100%` in my styles, it converts it to `100% 1 0%` which is not valid in Chrome. In `3.2.0` it converted that same style to `100% 1`

**Comment by luisrudge** [comment]
can you tell me why that's not valid? is that because you can't use `percentage` `number` `percentage` as a value?

**Comment by zslabs** [comment]
Hey Luís,
I couldn't find anything about "why" it doesn't work, but in the Chrome inspector, it blocks out that entire `flex` line as an invalid value for that property.

**Comment by luisrudge** [comment]
and 100% 1 is valid?

**Comment by luisrudge** [comment]
can you do a quick codepen so I can test it?

**Comment by zslabs** [comment]
https://codepen.io/zslabs/pen/BYymvP

**Comment by luisrudge** [comment]
There's something really weird going on on both chrome and firefox.

```css
.flex-child1 {
  flex: 100% 2;
  /*
  flex-basis: 100%;
  flex-grow: 2;
  flex-shrink: 1;
  */
}
```

But the spec says that with two values, the first one should be unitless:
![image](https://user-images.githubusercontent.com/941075/35579597-2c221e3a-05ce-11e8-8c87-f46c7a902696.png)

https://developer.mozilla.org/en-US/docs/Web/CSS/flex#Syntax

**Comment by Zephir77167** [comment]
Hey, after some testing it is the more apparent on Safari iOS <= 10

**Comment by luisrudge** [comment]
So, all browsers are not following the spec? 😝

**Comment by zslabs** [comment]
Are we kind of at an impasse with this one? The docs around this are a bit confusing; I just haven't found any Chromium bugs to back up this being an issue with how the browser is handling it.

**Comment by luisrudge** [comment]
I honestly have no idea. The spec is very clear, but, apparently, browsers are not following the spec at all :(

**Comment by judowalker** [comment]
Please correct me if I'm wrong, but isn't the actual spec here:
https://www.w3.org/TR/css-flexbox-1/#flex-property

According to that, the acceptable values are:
`none | [ <‘flex-grow’> <‘flex-shrink’>? || <‘flex-basis’> ]`

The double bar (||) separates two or more options: one or more of them must occur, **in any order**.
https://drafts.csswg.org/css-values-4/#comb-any

If I'm interpreting that correctly, that means `flex-basis` can be first. So Chrome is correct in treating `100% 1 0%` as invalid but `100% 1 0` as valid with `100%` being `flex-basis`.

**Comment by luisrudge** [comment]
You must be right. I only glanced the spec and didn't read the double || thing. I'm glad they're right and I'm wrong though!

**Comment by adrienharnay** [comment]
Yeah this is actually great news!

**Comment by adrienharnay** [comment]
Could we arrange a revert of 3.3.0 then?

**Comment by luisrudge** [comment]
fixed in https://github.com/luisrudge/postcss-flexbugs-fixes/pull/46

published in 3.3.1

Thanks everyone!

**Comment by adrienharnay** [comment]
Neat, thanks to you!
